include/constants/pokemon.h

```diff
@@ -243,71 +243,80 @@
-#define EVO_CRITICAL_HITS                 35     // Pokémon performs specified number of critical hits in one battle
-#define EVO_SCRIPT_TRIGGER_DMG            36     // Pokémon has specified HP below max, then player interacts trigger
-#define EVO_DARK_SCROLL                   37     // interacts with Scroll of Darkness
-#define EVO_WATER_SCROLL                  38     // interacts with Scroll of Waters
-#define EVO_ITEM_NIGHT                    39     // specified item is used on Pokémon, is night
-#define EVO_ITEM_DAY                      40     // specified item is used on Pokémon, is day
-#define EVO_ITEM_HOLD                     41     // Pokémon levels up, holds specified item
-#define EVO_LEVEL_FOG                     42     // Pokémon reaches the specified level during fog in the overworld
-#define EVO_MOVE_TWO_SEGMENT              43     // Pokémon levels up, knows specified move, has a personality value with a modulus of 0
-#define EVO_MOVE_THREE_SEGMENT            44     // Pokémon levels up, knows specified move, has a personality value with a modulus of 1-99
-#define EVO_LEVEL_FAMILY_OF_THREE         45     // Pokémon reaches the specified level in battle with a personality value with a modulus of 0
-#define EVO_LEVEL_FAMILY_OF_FOUR          46     // Pokémon reaches the specified level in battle with a personality value with a modulus of 1-99
-#define EVO_USE_MOVE_TWENTY_TIMES         47     // Pokémon levels up after having used a move for at least 20 times
-#define EVO_RECOIL_DAMAGE_MALE            48     // Pokémon levels up after having suffered specified amount of non-fainting recoil damage as a male
-#define EVO_RECOIL_DAMAGE_FEMALE          49     // Pokémon levels up after having suffered specified amount of non-fainting recoil damage as a female
-#define EVO_ITEM_COUNT_999                50     // Pokémon levels up after trainer has collected 999 of a specific item
-#define EVO_DEFEAT_WITH_ITEM              51     // Pokémon levels up after having defeat 3 Pokémon of the same species holding the specified item
-#define EVO_OVERWORLD_STEPS               52     // Pokémon levels up after having taken a specific amount of steps in the overworld
-
-// Evolution 'modes,' for GetEvolutionTargetSpecies
-#define EVO_MODE_NORMAL            0
-#define EVO_MODE_CANT_STOP         1
-#define EVO_MODE_TRADE             2
-#define EVO_MODE_ITEM_USE          3
-#define EVO_MODE_ITEM_CHECK        4 // If an Everstone is being held, still want to show that the stone *could* be used on that Pokémon to evolve
-#define EVO_MODE_BATTLE_SPECIAL    5
-#define EVO_MODE_OVERWORLD_SPECIAL 6
-#define EVO_MODE_BATTLE_ONLY       7 // This mode is only used in battles to support Tandemaus' unique requirement
+
+#if MODERN
+#define U8_ENUM : u8
+#else
+#define U8_ENUM
+#endif
```

Comment:
This should probably be in `global.h`

Reply:
I assume you mean `inlcude/global.h. Where should I add them there?`

Reply:
I think `include/constants/global.h` is probably better (needed for when `master` gets merged to `upcoming` and we're able to start using `enum`s in asm files).

Reply:
should I make defines for `U16_ENUM` and `U32_ENUM` as well?

Reply:
That's a good question! I'd be surprised if we'd ever be able to use a `U32_ENUM`—that's a lot of values. `U16_ENUM` seems useful though, things like species, abilities, moves, items, move effects (?), and dex numbers could use that... But I wouldn't be opposed to doing that when needed rather than up-front, I'll leave the decision to you/Edu :)

Reply:
I've added the u16 (+u8) in `include/constants/global.h` but not opposed to removing it for now

Reply:
It doesn't compile anymore

Reply:
Hmm that's really interesting...

The error looks like:
```
include/constants/pokemon.h: Assembler messages:
include/constants/pokemon.h:249: Error: bad instruction `enum EvolutionMethods U8_ENUM {'
include/constants/pokemon.h:250: Error: bad instruction `evo_friendship_day,'
include/constants/pokemon.h:251: Error: bad instruction `evo_friendship,'
...
include/constants/pokemon.h:312: Error: bad instruction `evo_mode_battle_only,'
include/constants/pokemon.h:313: Error: junk at end of line, first unrecognized character is `}'
```
Which I'm struggling to understand. I don't see why this wouldn't work, but the previous commit would. Maybe `include/constants/global.h` isn't included? Maybe the `preproc`-based `enum` handling can't cope with this `U8_ENUM` thing being defined in a separate file? :thinking:

Reply:
It looks like it doesn't work if it is in a different file. I think for now we will have to include it in the same file. For as long as agbcc is still there at least. Later we can just get rid of the define and use it a u8/u16 directly.

Reply:
Shouldn't we just label this 1.10 then and wait with integrating it? A more global enum conversion can be part of 1.10, which won't support agbcc anymore anyway. Feels like we're only giving ourselves more work by accounting for agbcc now and then cleaning it up later when we can just wait

Reply:
that's fair